### PR TITLE
Add setdefault option to tcl module

### LINKED ISSUE #13714: Add setdefault option to tcl environment-modules backend
Spack Tcl environment-modules backend does not have the `setdefault` version as the LMod backend does.
So, when one installs an application and uses the Tcl environment-modules backend, it is left to the environment-modules to pick the default. This may change the current default on the system, which is not ideal.

For completeness
```
spack module lmod -h
usage: spack module lmod [-h] SUBCOMMAND ...

positional arguments:
  SUBCOMMAND
    refresh   regenerate module files
    find      find module files for packages
    rm        remove module files
    loads     prompt the list of modules associated with a constraint
    setdefault
              set the default module file for a package

optional arguments:
  -h, --help  show this help message and exit
```

For the Tcl backend...
```
spack module tcl -h
usage: spack module tcl [-h] SUBCOMMAND ...

positional arguments:
  SUBCOMMAND
    refresh   regenerate module files
    find      find module files for packages
    rm        remove module files
    loads     prompt the list of modules associated with a constraint

optional arguments:
  -h, --help  show this help message and exit
```

### Rationale

Assuming that there is no other GROMACS installation on the system if one uses the Tcl environment-modules backend and he/she performs the operation
```
spack install gromacs@2019.2
```
The "default" module version will be `gromacs/2019.2`. 
After performing the operation
```
spack install gromacs@2018.3
```
The default module version will be `gromacs/2018.3`. 

This is a problem on the HPC centers because care has to be taken when changing default module versions. For this reason, the LMod backend has this option. But this option is also needed by centers that do not use LMod.

### Description

The solution is to add the option to the Tcl backend. I haven't looked at how to implement it yet.

### PR DESCRIPTION
**Pull request by victorusu**
This PR introduces the command

```
spack module tcl setdefault <package>
```

similar to the lmod command.

Closes #13714

### CODE REVIEW

**Comment by adamjstewart** [comment]
Ping @alalazo

**Comment by alalazo** [comment]
@victorusu Can you rebase and ping me? (Apologies for the delay, I'm going through old PRs and check what is still relevant)

**Comment by alalazo** [comment]
@victorusu

**Comment by victorusu** [comment]
@alalazo, done! Sorry

**Comment by alalazo** [comment]
@victorusu Can you fix style?

File: lib/spack/spack/cmd/modules/tcl.py [hunk 1]
```diff
@@ -4,16 +4,46 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 import functools
+import os
 
+import llnl.util.filesystem
+import spack.cmd.common.arguments
```
Comment (tldahlgren) [inline lib/spack/spack/cmd/modules/tcl.py]: The following *should* fix the style issue:

```suggestion
import llnl.util.filesystem

import spack.cmd.common.arguments
```

**Review by tldahlgren — DISMISSED** [review]

**Review by tldahlgren — COMMENTED** [review]
Merged style fix for expedience since this PR is in the list for the next point release.

**Review by alalazo — APPROVED** [review]

**Review by haampie — APPROVED** [review]

**Pull request merged**